### ISSUE TITLE
OpenContainer (from animation package) RootNavigator not working in GetMaterialApp.router

### ISSUE THREAD

**Issue by AbdallahR99** [jonataslaw/getx#1824]
I want to change the current url on click OpenContainer

```
OpenContainer(
              useRootNavigator: true,
              routeSettings: const RouteSettings(
                name: Routes.Details,
              ),
...
```

the URL is not changed on open container

and that problem is happen when using GetMaterialApp.router 

can fix that problem please?

or how to change the current route manually? 

flutter verison: 2.2.3
get: ^4.3.8
flutter doctor:
> Doctor summary (to see all details, run flutter doctor -v):
> [√] Flutter (Channel unknown, 2.2.3, on Microsoft Windows [Version 10.0.19043.1151], locale en-US)
> [√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
> [√] Chrome - develop for the web
> [√] Android Studio (version 4.1.0)
> [√] VS Code (version 1.59.1)
> [√] Connected device (3 available)
> 
> • No issues found!

It is prefer that there be documentation about the GetMaterialApp.router

**Comment by FlawLessx** [comment]
> I want to change the current url on click OpenContainer
> 
> ```
> OpenContainer(
>               useRootNavigator: true,
>               routeSettings: const RouteSettings(
>                 name: Routes.Details,
>               ),
> ...
> ```
> 
> the URL is not changed on open container
> 
> and that problem is happen when using GetMaterialApp.router
> 
> can fix that problem please?
> 
> or how to change the current route manually?
> 
> flutter verison: 2.2.3
> get: ^4.3.8
> flutter doctor:
> 
> > Doctor summary (to see all details, run flutter doctor -v):
> > [√] Flutter (Channel unknown, 2.2.3, on Microsoft Windows [Version 10.0.19043.1151], locale en-US)
> > [√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
> > [√] Chrome - develop for the web
> > [√] Android Studio (version 4.1.0)
> > [√] VS Code (version 1.59.1)
> > [√] Connected device (3 available)
> > • No issues found!
> 
> It is prefer that there be documentation about the GetMaterialApp.router

did you found a solution?

**Comment by jonataslaw** [comment]
I still don't understand where this problem relates to this library. Could you please be more clear?
From what I could research, OpenContainer is part of a third-party library maintained by google, and it appears to use navigator 1 (https://github.com/flutter/packages/blob/master/packages/animations/lib/src/open_container.dart #L283)
I can't see where this relates to GetX.

**Comment by FlawLessx** [comment]
> I still don't understand where this problem relates to this library. Could you please be more clear?
> From what I could research, OpenContainer is part of a third-party library maintained by google, and it appears to use navigator 1 (https://github.com/flutter/packages/blob/master/packages/animations/lib/src/open_container.dart#L283)
> I can't see where this relates to GetX.

Yeah you're right, i can reproduce this and works fine with get route system. I think you can close this

**Comment by AbdallahR99** [comment]
@jonataslaw not just openContainer 
even if you used dialog with ``useRootNavigator`` the URL is not changing 
I can't find a way to set the current URL manually

**Comment by AbdallahR99** [comment]
@FlawLessx do you use Navigator 2.0?

**Comment by AbdallahR99** [comment]
@FlawLessx 
if it's works with you then what's the wrong with my code?

https://github.com/AbdallahR99/nav2issue/blob/main/lib/main.dart

**Comment by FlawLessx** [comment]
@AbdallahR99 i think the problem is you're using getmaterialapp.router, here is what i'm using https://gist.github.com/FlawLessx/698396fd6cb4868f9c8acac3e20ac84d

**Comment by FlawLessx** [comment]
> @FlawLessx
> if it's works with you then what's the wrong with my code?
> 
> https://github.com/AbdallahR99/nav2issue/blob/main/lib/main.dart

what i can tell you is you want to navigate to page with named routes '/page2', but that route is not even registered in getPages

**Comment by AbdallahR99** [comment]
@FlawLessx 

> > @FlawLessx
> > if it's works with you then what's the wrong with my code?
> > https://github.com/AbdallahR99/nav2issue/blob/main/lib/main.dart
> 
> what i can tell you is you want to navigate to page with named routes '/page2', but that route is not even registered in 

it's not working even if i registered it
